pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -547,19 +547,9 @@ func (m *managerImpl) evictPod(pod *v1.Pod, gracePeriodOverride int64, evictMsg
 	// If the pod is marked as critical and static, and support for critical pod annotations is enabled,
 	// do not evict such pods. Static pods are not re-admitted after evictions.
 	// https://github.com/kubernetes/kubernetes/issues/40573 has more details.
-	if kubelettypes.IsStaticPod(pod) {
-		// need mirrorPod to check its "priority" value; static pod doesn't carry it
-		if mirrorPod, ok := m.mirrorPodFunc(pod); ok && mirrorPod != nil {
-			// skip only when it's a static and critical pod
-			if kubelettypes.IsCriticalPod(mirrorPod) {
-				klog.Errorf("eviction manager: cannot evict a critical static pod %s", format.Pod(pod))
-				return false
-			}
-		} else {
-			// we should never hit this
-			klog.Errorf("eviction manager: cannot get mirror pod from static pod %s, so cannot evict it", format.Pod(pod))
-			return false
-		}
+	if kubelettypes.IsCriticalPod(pod) {
```

Comment:
From the discussion of https://github.com/kubernetes/kubernetes/issues/73572, we got that, In `IsCriticalPod --> IsCriticalPodBasedOnPriority(*pod.Spec.Priority)`, the func should use `pod.Spec.Priority` to do something.
But if users only set `spec.priorityClassName` rather than `spec.priority` to a static pod, the admission-controller only set  `spec.priority` to the mirror pod, and the static pod won't have  `spec.priority`.
So, it seems that, for static pod, we should send its mirror pod to the func  IsCriticalPod.